### PR TITLE
Implemented business scenario for airline FlyAir.

### PR DESCRIPTION
**Pull request by Chanamindy**
I did the work, please review and let me know if anything needs to be changed.

### CODE REVIEW

File: 3 reports airline.sql [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+from Flight f
+group by f.FlightNumber, f.DepartureAirport, f.TimeDeparting, f.ArrivalAirport
+
+--2)
+select datediff(day,(getdate()),(TimeDeparting)), *
+from Flight f 
+where f.PassportNumber is null and datediff(day,(getdate()),(TimeDeparting)) between 0 and 7
+
+--3)
+select AmountDeparting = count(distinct f.FlightNumber), DepartureDate = datefromparts(year(TimeDeparting), month(TimeDeparting), day(TimeDeparting)), AmountOfPassengers = count(f.FlightNumber)
```
Comment (AvigailGellis) [inline 3 reports airline.sql:12]: It isn't necessary to do a datefromparts here, just use the timedeparting column alone.

File: 3 reports airline.sql [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+from Flight f 
+group by datefromparts(year(f.TimeDeparting), month(f.TimeDeparting), day(f.TimeDeparting))
+
+--4)
```
Comment (AvigailGellis) [inline 3 reports airline.sql:16]: Destination means the arrival country, not airport.

**Review by AvigailGellis — CHANGES_REQUESTED** [review]
Amazing job! See comments and resubmit.

File: 1 table airline.sql [hunk 3]
```diff
@@ -0,0 +1,33 @@
+use FlyAirDB
+go 
+drop table if exists Flight
+go 
+create table dbo.Flight(
+    FlightId int not null identity primary key,
+    FlightNumber varchar(6) not null constraint ck_Flight_flight_number_starts_with_fly_then_3_digits_starting_from_001 check(FlightNumber like 'FLY[0-9][0-9][1-9]'),
```
[the remark this answers is not in the document]
Comment (Chanamindy) [reply 1 table airline.sql]: Changed it.

**Comment by AvigailGellis** [comment]
Is this ready for review or you are still in middle of making corrections?

**Comment by Chanamindy** [comment]
Still making more changes. I dropped the files into the pull request so how do I bring down the comments so I can make changes on my local computer?

**Comment by Chanamindy** [comment]
"This constraint has to be combined with the constraint about if a passport can be used to fly. Also, don't the instructions say it's valid for 9 years and 6 months? Not 10 years..."

Is the constraint for if it can be used to fly good? All I need to do is combine it with the constraint that it's only valid for 9y and 6m?

The rest of my comments can be reviewed.  I only need to fix this constraint.

**Comment by AvigailGellis** [comment]
See comment above.

**Comment by Chanamindy** [comment]
My changes are ready to be marked.

**Comment by AvigailGellis** [comment]
Sorry for the confusion, please see comment and resubmit. The spec is not crazy clear on this part so it's a bit of a challenge but follow the instructions I wrote and then you will be done.

**Comment by Chanamindy** [comment]
I don't see the comment. Where is it?

**Comment by AvigailGellis** [comment]
image

**Comment by Chanamindy** [comment]
Please mark my work with the updated constraints.
Thank you.

**Comment by AvigailGellis** [comment]
You split it into 2 separate constraints, it must be combined into 1 with or, otherwise it will crash when it doesn't follow one of them.

**Comment by Chanamindy** [comment]
I'm already so mixed up which two constraints should I combine with a or?

**Comment by AvigailGellis** [comment]
yes, sorry this spec wasn't so clear about what was needed.

**Comment by Chanamindy** [comment]
I combined the two constraints.

**Comment by AvigailGellis** [comment]
Great job! Sorry it took so long!

**Pull request closed without merging**